### ISSUE TITLE
Some bug(?) disables the mod mid-game

### ISSUE THREAD

**Issue by originalfoo** [boformer/LegacyNetworkSkins#17]
It's happened few times now: Mid-game, something happens and all networks revert to their factory default lights, etc., and the mod's UI disappears, only becoming visible again after exit to desktop and reload. It's literally like the mod was unloaded mid-game.

I suspect it might be something to do with lack of RAM, particularly if I Alt-Tab to another app while game is running. There's message in the log about serialised XML files being purged or something?

When the issue occurs it also seems to fubar Surface Painter (mod is still operational, but all existing painted surfaces tend to revert to whatever they were before being painted).

I managed to upload a savegame shortly after the issue occurred, along with a player.log:

* Save: https://steamcommunity.com/sharedfiles/filedetails/?id=1396655467
* Log: https://drive.google.com/open?id=11LMpcI9ALEPUtFv4x8EiGg6svtzCEuMf (too big for pastebin due to spammy MOM error)

**Comment by originalfoo** [comment]
Just noticed this issue about max data size in the TM:PE repo:

https://github.com/VictorPhilipp/Cities-Skylines-Traffic-Manager-President-Edition/issues/160

> Custom data saving/loading should be optimized regarding memory consumption because custom mod data must be represented with less than 16711680 bytes (~16 MB, see ColossalFramework.IO.DataSerializer.WriteByteArray) and because the deserialization process might take a long time.

Could something similar be causing the issue I mention above?

I've done some more testing and the issue only seems to occur when I've skinned thousands of segments (eg. vast highway networks and lots of small rural roads which I use extensively). Without skinning my vast highway network, I've noticed the problem does not occur which leads me to think it might be some sort of data overflow issue?
